### PR TITLE
BAH-1013 | Events raised for recurring appointments and fix weekly recurring appointment

### PR DESCRIPTION
**Pull request by venkatavineela**

### CODE REVIEW

**Comment by bsneha90** [comment]
@venkatavineela @binduak we need to understand what change we are making in product for this to work. Product does not have mart officially. So please mention in JIRA card what change are we trying to make, example raise a particular atom feed event, etc.

File: api/src/main/java/org/openmrs/module/appointments/advice/RecurringAppointmentsAdvice.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    private static final String CHANGE_STATUS = "changeStatus";
+    private static final List<String> METHOD_NAMES = Arrays.asList(VALIDATE_AND_SAVE, UPDATE, CHANGE_STATUS);
+    private static final String URL_PATTERN_GLOBAL_PROPERTY = "atomfeed.event.urlPatternForRecurringAppointments";
+    private static final String DEFAULT_URL_PATTERN = "/openmrs/ws/rest/v1/recurring-appointments?uuid={uuid}";
+
+    @Override
+    public void afterReturning(Object returnValue, Method method, Object[] arguments, Object target) throws Throwable {
+        if (VALIDATE_AND_SAVE.equals(method.getName()) || isAllRecurringAppointmentsUpdate(method, returnValue)) {
+            AppointmentRecurringPattern appointmentRecurringPattern = (AppointmentRecurringPattern) returnValue;
+            raiseEventsForAppointments(method, arguments, target, new ArrayList<>(appointmentRecurringPattern.getAppointments()));
```
Comment (bsneha90) [inline api/src/main/java/org/openmrs/module/appointments/advice/RecurringAppointmentsAdvice.java:29]: raiseEventsForAppointments always is called, only what list of appointments we pass changes. Can we bring it out commonly?

File: omod/src/main/java/org/openmrs/module/appointments/web/service/impl/DailyRecurringAppointmentsGenerationService.java [hunk 2]
```diff
@@ -61,7 +60,11 @@ public Date getEndDate(int period, Integer frequency, Date endDate) {
     public List<Appointment> addAppointments(AppointmentRecurringPattern appointmentRecurringPattern,
                                              RecurringAppointmentRequest recurringAppointmentRequest){
         this.recurringAppointmentRequest = recurringAppointmentRequest;
-        List<Appointment> appointments = appointmentRecurringPattern.getAppointments().stream().collect(Collectors.toList());
+        List<Appointment> activeAppointments = new ArrayList<>(appointmentRecurringPattern.getActiveAppointments());
```
Comment (bsneha90) [inline omod/src/main/java/org/openmrs/module/appointments/web/service/impl/DailyRecurringAppointmentsGenerationService.java:63]: Why are we changing this as Part of this PR?

File: api/src/main/java/org/openmrs/module/appointments/advice/RecurringAppointmentsAdvice.java (hunk 1 above)
Comment (binduak) [reply api/src/main/java/org/openmrs/module/appointments/advice/RecurringAppointmentsAdvice.java]: Done !

File: omod/src/main/java/org/openmrs/module/appointments/web/service/impl/DailyRecurringAppointmentsGenerationService.java (hunk 2 above)
Comment (binduak) [reply omod/src/main/java/org/openmrs/module/appointments/web/service/impl/DailyRecurringAppointmentsGenerationService.java:62]: @bsneha90 As part of raising events for recurring appointments, we had to consider the voided appointments in the bridge table. The changes are specific to that !

**Comment by binduak** [comment]
@bsneha90 Updated the JIRA card with the documentation. Please check

**Pull request merged**